src/postgres/tree/PostgresServerTreeItem.ts

```diff
@@ -73,4 +75,40 @@ export class PostgresServerTreeItem extends AzureParentTreeItem<ISubscriptionCon
                 return false;
         }
     }
+
+    public async checkAndConfigureFirewall(): Promise<void> {
+        const firewallRuleName: string = 'azureDatabasesForVSCode-publicIp';
+        const resourceGroup: string = azureUtils.getResourceGroupFromId(this.id);
+        const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
+        const serverName: string = nonNullProp(this.server, 'name');
+        const ip: string = await publicIp.v4();
+        let existingFirewallRule: FirewallRule | undefined;
+
+        // Creating/updating a firewall rule takes a long time. So check if the rule already exists and is valid before creating/updating
+        try {
+            existingFirewallRule = await client.firewallRules.get(resourceGroup, serverName, firewallRuleName);
+        } catch (error) {
+            if (parseError(error).errorType === 'ResourceNotFound') {
+                // This firewall rule is not configured yet. Ignore error
+            } else {
+                throw error;
+            }
+        }
+
+        if (!existingFirewallRule || existingFirewallRule.startIpAddress !== ip || existingFirewallRule.endIpAddress !== ip) {
+            const newFirewallRule: FirewallRule = {
+                startIpAddress: ip,
+                endIpAddress: ip
+            };
+
+            const options: vscode.ProgressOptions = {
+                location: vscode.ProgressLocation.Notification,
+                title: localize('configuringFirewall', 'Configuring firewall for server "{0}"...', serverName)
```

Comment:
I would prompt first before doing something like this since it's security related and could take a while